Sources/SwiftySensors/Sensor.swift

```diff
@@ -275,7 +275,7 @@ extension Sensor: CBPeripheralDelegate {
     
     /// :nodoc:
     public func peripheral(_ peripheral: CBPeripheral, didUpdateValueFor characteristic: CBCharacteristic, error: Error?) {
-        guard let service = services[characteristic.service.uuid.uuidString] else { return }
+        guard let service = services[characteristic.service?.uuid.uuidString ?? ""] else { return }
```

Comment:
Just a thought but would it logically make more sense to use guard to check for the service/uuidString instead of defaulting to an empty string?

```
guard let uuidString = characteristic.service?.uuid.uuidString, let service = services[uuidString] else { return }
```

Functionally I think they're about the same but it seems weird you might try to access a service with the uuid of "".

Reply:
Did the service param change from an optional? you can't access a map value with an optional key, hence the ""

Reply:
This function is a callback from CoreBluetooth.  You're already indexing services by uuid.  If for some reason CB sends you an update for a service without a UUID, you're basically screwed, so, I'd say ignore it.

```
        guard let serviceUUID = characteristic.service?.uuid.uuidString else { return }
        guard let service = services[serviceUUID] else { return }
```